main/algorithm_api_mock.h

```diff
@@ -0,0 +1,75 @@
+#ifndef MAIN_ALGORITHM_API_MOCK_H
+#define MAIN_ALGORITHM_API_MOCK_H
+
+#include <maze_solver/direction.h>
+#include <misc_utils/physical_size.h>
+
+#include "position.h"
+#include "temp_map.h"
+
+#include <array>
+#include <optional>
+
+namespace micromouse
+{
+
+constexpr Position convert(std::size_t row, std::size_t col, Direction heading)
```

Comment:
```suggestion
constexpr Position convert(std::size_t row, std::size_t col, Direction heading) noexcept
```

(Or ignore because I'll edit this with the new version anyway)